src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorFormattingPass.cs

```diff
@@ -247,5 +364,69 @@ private static void ShrinkToSingleSpace(SyntaxNode node, List<TextEdit> edits, R
             };
             edits.Add(edit);
         }
+
+        private static bool FormatBlock(FormattingContext context, RazorSourceDocument source, SyntaxNode openBraceNode, SyntaxNode codeNode, SyntaxNode closeBraceNode, IList<TextEdit> edits, bool additionalIndentation)
+        {
+            var didFormat = false;
+
+            var openBraceRange = openBraceNode.GetRangeWithoutWhitespace(source);
+            var codeRange = codeNode.GetRangeWithoutWhitespace(source);
+            if (openBraceRange is not null &&
+                codeRange is not null &&
+                openBraceRange.End.Line == codeRange.Start.Line &&
+                // Because we don't always know what kind of Razor object we're operating on we have to do this to avoid duplicate edits.
+                // The other way to accomplish this would be to apply the edits after every node and function, but that's not in scope for my current work.
+                !edits.Any(e => e.Range.End == codeRange.End))
```

Comment:
I'd pull this and the comment above into a helper method for verbosity. Also, out of sheer curiosity do we know of examples where edits already exist after the opening brace?

Reply:
Done on the HelperMethod. Wouldn't have bothered doing that speculatively, `FormatsShortBlock` fails without this check.